apps/challenges/views.py

```diff
@@ -340,3 +343,56 @@ def challenge_phase_split_list(request, challenge_pk):
     serializer = ChallengePhaseSplitSerializer(result_page, many=True)
     response_data = serializer.data
     return paginator.get_paginated_response(response_data)
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['GET'])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_all_submissions_of_challenge(request, challenge_pk, challenge_phase_pk):
+    """
+    Returns all the submissions for a particular challenge
+    """
+    # To check for the corresponding challenge from challenge_pk.
+    challenge_model = get_model_object(Challenge)
+    challenge = challenge_model(challenge_pk)
+
+    # To check for the corresponding challenge phase from the challenge_phase_pk and challenge.
+    try:
+        challenge_phase = ChallengePhase.objects.get(pk=challenge_phase_pk, challenge=challenge)
+    except ChallengePhase.DoesNotExist:
+        response_data = {'error': 'Challenge Phase {} does not exist'.format(challenge_phase_pk)}
+        return Response(response_data, status=status.HTTP_404_NOT_FOUND)
+
+    # To check for the user as a host of the challenge from the request and challenge_pk.
+    if is_user_a_host_of_challenge(user=request.user, challenge_pk=challenge_pk):
```

Comment:
A slightly better way of doing this can be catch result of `is_user_a_host_of_challenge` and `has_user_participated_in_challenge` in appropriate variable before. Then first go for the else condition where in you are returning 400. So this will look something like this

```
if not is_challenge_host and not is_participant:
 # return 400 one here
else:
  if is_challenge_host:
    # form submission_params here 
    submission_params['challenge_phase__challenge'] = challenge.

```

Also `ChallengeSubmissionManagementSerializer` and `SubmissionSerializer` can be easily catched in a variable and then used accordingly.

This is just a suggestion. Let me know if you find any difficulty in this or if this is not suitable.